src/ui/wxWidgets/PasswordSafeFrame.cpp

```diff
@@ -1389,6 +1389,9 @@ int PasswordSafeFrame::Open(const wxString &fname)
   if (rc != PWScore::SUCCESS)
     return rc;
 
+  // Clear the lock on the old file, if any.
+  m_core.SafeUnlockCurFile();
```

Comment:
I’m thinking maybe we should wait until after the password check - maybe put this around line 1444 (ClearAppData())?

Reply:
SafeUnlockCurFile() calls GetCurFile() to get the name of the lock file, so it has to be called before SetCurFile() at line 1402 (setting the name of the new file to be opened.)  I'll add a comment to that effect.  The areas you mention, ClearAppData() and line 1444, are within a '#if 0' block.  I don't know the history of that code, but checking the PW seems to be done elsewhere.

Reply:
What about the return condition on line 1412 - if we get there, the password didn't match, and seems like the current file should remain open and locked?

Reply:
I see what you mean now, you're right.  I added a new version of the SafeUnlockFile function so I could save the file name and pass it later.  Please, have another look.  Thanks for checking this.